### PR TITLE
Java: Add `FUNCTION DUMP` and `FUNCTION RESTORE`.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
https://redis.io/docs/latest/commands/function-dump/
https://redis.io/docs/latest/commands/function-restore/

TODO:
- [x] Docs
- [x] Manual tests
- [x] UT
- [x] standalone client IT
- [ ] Merge https://github.com/amazon-contributing/redis-rs/pull/159
- [ ] cluster client IT (depends on ^)

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsClusterCommands.java [hunk 1]
```diff
@@ -266,6 +269,111 @@ CompletableFuture<ClusterValue<Map<String, Object>[]>> functionList(
      */
     CompletableFuture<String> functionDelete(String libName, Route route);
 
+    /**
+     * Returns the serialized payload of all loaded libraries.<br>
+     * The command will be routed to all primary nodes.
+     *
+     * @since Redis 7.0 and above.
+     * @see <a href="https://redis.io/docs/latest/commands/function-dump/">redis.io</a> for details.
+     * @return The serialized payload.
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsClusterCommands.java:278]: might be slightly clearer although I know we already stated this above, up to you. Same applies elsewhere in this PRs docs
```suggestion
     * @return The serialized payload of all loaded libraries.
```

File: java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsClusterCommands.java [hunk 2]
```diff
@@ -266,6 +269,111 @@ CompletableFuture<ClusterValue<Map<String, Object>[]>> functionList(
...
+     * @example
+     *     <pre>{@code
+     * ClusterValue<byte[]> data = client.functionDump().get();
+     * // data stores serialized dump from all primary nodes
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsClusterCommands.java:282]:
```suggestion
     * // data contains the serialized dump from all primary nodes
```

File: java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsClusterCommands.java [hunk 3]
```diff
@@ -266,6 +269,111 @@ CompletableFuture<ClusterValue<Map<String, Object>[]>> functionList(
...
+     * // it could be saved to restore loaded functions on any Redis instance
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsClusterCommands.java:283]: to be consistent with the other docs:
```suggestion
     * // now data could be saved to restore loaded functions on any Redis instance
```

File: java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsClusterCommands.java [hunk 4]
```diff
@@ -266,6 +269,111 @@ CompletableFuture<ClusterValue<Map<String, Object>[]>> functionList(
...
+     * @example
+     *     <pre>{@code
+     * byte[] data = client.functionDump(RANDOM).get().getSingleValue();
+     * // now data could be saved to restore loaded functions on any Redis instance
+     * }</pre>
+     */
+    CompletableFuture<ClusterValue<byte[]>> functionDump(Route route);
+
+    /**
+     * Restores libraries from the serialized payload.<br>
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsClusterCommands.java:305]: might be slightly clearer although I know we already state this below, up to you. Same applies elsewhere in this PRs docs:
```suggestion
     * Restores libraries from the serialized payload returned by {@link #functionDump()}.<br>
```

File: java/client/src/main/java/glide/api/BaseClient.java [hunk 5]
```diff
@@ -303,19 +303,24 @@ protected static CommandManager buildCommandManager(ChannelHandler channelHandle
...
-     * than also returns <code>null</code>.
+     * exception or returns the value as an object of type <code>T</code>.
      *
      * @param response Redis protobuf message.
      * @param classType Parameter <code>T</code> class type.
-     * @param isNullable Accepts null values in the protobuf message.
+     * @param flags A set of parameters which describes how to handle the response. Could be empty or
+     *     any combination of
+     *     <ul>
+     *       <li>{@link ResponseFlags#ENCODING_UTF8} to return string data as a <code>byte[]</code>
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/BaseClient.java:313]: Oh it works in vice versa, fixing the doc

File: java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsCommands.java [hunk 6]
```diff
@@ -128,6 +129,53 @@ public interface ScriptingAndFunctionsCommands {
...
+     * @example
+     *     <pre>{@code
+     * String response = client.functionRestore(data).get();
+     * assert response.equals("OK");
+     * }</pre>
+     */
+    CompletableFuture<String> functionRestore(byte[] payload);
+
+    /**
+     * Restores libraries from the serialized payload.
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsCommands.java:163]:
```suggestion
     * Restores libraries from the serialized payload returned by {@link #functionDump()}.
```

**Review by aaron-congo — APPROVED** [review]

**Pull request merged**